src/App.tsx

```diff
@@ -1,24 +1,233 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+/* eslint-disable no-param-reassign */
+import React, {
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  createTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { AuthContext } from './components/Auth/AuthContext';
+import { NewTodoField } from './components/NewTodoField/NewTodoField';
+import { TodoFilters } from './components/TodoFilters/TodoFilters';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import {
+  ErrorNotification,
+} from './components/ErrorNotification/ErrorNotification';
+import { ErrorType } from './types/ErrorType';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [isUploadError, setIsUploadError] = useState(false);
+  const [error, setError] = useState(ErrorType.None);
+  const [isError, setIsError] = useState(false);
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filtredTodos, setFiltredTodos] = useState<Todo[]>([]);
+  const [isTodoLoading, setIsTodoLoading] = useState(false);
+  const [deletedTodosId, setDeletedTodosId] = useState<number[]>([]);
+  const [activeTodoId, setActiveTodoId] = useState<number[]>([]);
+  const [updatedTodoID, setUpdatedTodoID] = useState<number[]>([]);
+
+  const user = useContext(AuthContext);
+
+  const loadTodosFromServer = async () => {
+    try {
+      if (user) {
+        const getTodosFromServer = await getTodos(user.id);
+
+        setTodos(getTodosFromServer);
+      }
+    } catch (e) {
+      setError(ErrorType.LoadError);
+      setIsError(true);
+    }
+  };
+
+  const uploadTodosOnServer = async (todo: Todo) => {
+    if (user) {
+      setIsTodoLoading(true);
+
+      try {
+        setActiveTodoId([0]);
+        setTempTodo(todo);
+        const newTodo = await createTodo({
+          title: todo.title,
+          userId: todo.userId,
+          completed: false,
+        });
+
+        setTempTodo(null);
+        setTodos(prevTodos => [...prevTodos, newTodo]);
+        setActiveTodoId([]);
+        setIsTodoLoading(false);
+      } catch {
+        setIsUploadError(true);
+        setTodos(todos);
+        setTempTodo(null);
+        setIsTodoLoading(false);
+        setError(ErrorType.UploadError);
+        setIsError(true);
+      }
```

Comment:
this code can be simplified by using `finally`
```suggestion
        setTodos(prevTodos => [...prevTodos, newTodo]);
        setActiveTodoId([]);        
      } catch {
        setError(ErrorType.UploadError);
      } finally {
        setTempTodo(null);      
        setIsTodoLoading(false);      
      }
```